packages/eslint-plugin/configs/recommended.js

```diff
@@ -1,39 +1,3 @@
 module.exports = {
-	parser: 'babel-eslint',
-	extends: [
-		require.resolve( './jsx-a11y.js' ),
-		require.resolve( './custom.js' ),
-		require.resolve( './react.js' ),
-		require.resolve( './esnext.js' ),
-	],
-	env: {
-		node: true,
-	},
-	globals: {
-		window: true,
-		document: true,
-		wp: 'readonly',
-	},
-	overrides: [
-		{
-			// Unit test files and their helpers only.
-			files: [
-				'**/@(test|__tests__)/**/*.js',
-				'**/?(*.)test.js',
-			],
-			extends: [
-				require.resolve( './test-unit.js' ),
-			],
-		},
-		{
-			// End-to-end test files and their helpers only.
-			files: [
-				'**/specs/**/*.js',
-				'**/?(*.)spec.js',
-			],
-			extends: [
-				require.resolve( './test-e2e.js' ),
-			],
-		},
-	],
+	extends: [ require.resolve( './recommended-with-formatting.js' ), 'eslint-config-prettier' ],
```

Comment:
Apologies for the lateness of this here, by removing the `recommended` config here and switching to `recommended-with-formatting` this may be troublesome for projects currently consuming `@wordpress/scripts` and using the ESLint config.

Will this affect those users in that the ESLint config is no longer available and they are not yet ready to switch to or know anything about Prettier?

p.s. I've not had the ability to study this use case in depth

Reply:
@ntwb The `recommended` config is still available. This is how it works after this patch:
- the `recommended-with-formatting` config includes all the rules that check code formatting, and that sometimes conflict with Prettier conventions. It's basically the former `recommended` config with a new name. It's intended for projects that want to opt-out from Prettier and continue to use ESLint for formatting.
- the `recommended` config continues to exist, and it extends the `recommended-with-formatting` config, and disables all the code formatting rules by applying `eslint-config-prettier`. Therefore, the new default from now is to _not_ check code formatting with ESLint, and use solely Prettier for code formatting.

> Will this affect those users in that the ESLint config is no longer available and they are not yet ready to switch to or know anything about Prettier?

If you just upgrade to the new version of `@wordpress/scripts` and do nothing else, ESLint will stop reporting code formatting issues. Until you add Prettier to your project explicitly, not tool will be handing code formatting.

Reply:
These changes should also be noted in the changelog

I'm still not sure on the idea of removing the ability for users to be able to format their code, need to think on this some more, I'm fully behind the switch to Prettier so by not doing this it may help in this regard and nudge users to add Prettier once they discover their code is no longer formatted 😏

----

Also, just looking at the code of the two files, instead of creating the _new_ `eslint-plugin/configs/recommended.js` config file, why not leave the existing `eslint-plugin/configs/recommended-with-formatting.js` fille as is and add `'eslint-config-prettier'` to `extends` in `recommended.js`?

There are no other changes to the overall ESLint config besides this one, and if it is to be the default anyway, thoughts?

e.g.:

```js
	extends: [
		'eslint-config-prettier'
		require.resolve( './jsx-a11y.js' ),
		require.resolve( './custom.js' ),
		require.resolve( './react.js' ),
		require.resolve( './esnext.js' ),
	],
```

Reply:
> Also, just looking at the code of the two files, instead of creating the _new_ `eslint-plugin/configs/recommended-with-formatting.js` config file, why not leave the existing `eslint-plugin/configs/recommended.js` fille as is and add `'eslint-config-prettier'` to `extends` in `recommended.js`?

@ntwb I'm not sure I understand your comment -- I believe this is exactly what I'm doing in this PR. Rename `recommended.js` to `recommended-with-formatting.js` with the content not changed, and then create a _new_ `recommended.js` that extends the `-with-formatting` one. The Git diff doesn't show the change as rename, but it's what happened.

Reply:
> These changes should also be noted in the changelog

Added an entry to `packages/eslint-plugin/CHANGELOG.md`

Reply:
Thanks, let me rephrase it, why create the new `recommended-with-formatting.js` file?

It's not required, the only change required is adding `eslint-config-prettier` to `extends` in the original `recommended.js` file.

I have created this PR against this branch demonstrating the using ESLint `--print-config`

> https://github.com/WordPress/gutenberg/pull/19744

Reply:
> why create the new `recommended-with-formatting.js` file?

Oh, I see the point now. In the November 19th core-js chat (Slack link [here](https://wordpress.slack.com/archives/C5UNMSU4R/p1574173838080600), no blog post summary found for that week), we discussed a proposal by @aduth to support two ESLint modes:
- one doesn't run the code formatting rules and let's Prettier completely take care of that. That's the new `recommended` behavior.
- the second still runs the code formatting roles, as it did until now, for projects that for some reason want to opt out from Prettier and still use `wp-scripts`. That's the `recommended-with-formatting` config. It intentionally provides an escape hatch to use the old convention.

Reply:
Right, in that case `eslint-config-prettier` should be in the `recommended-with-formatting.js` file, not the `recommended.js` file because as it stands both configurations have identical configurations at the moment.

Reply:
It's all good as is according to what we agreed during the core-js chat. `recommended-with-formatting` is what we had so far where ESLint does some code styling checks, whereas `recommended` has many of the rules disables and lets Prettier own it with formatting.

Reply:
> I think I have now fully confused myself, I'll come back to this later today

Yes, there are at least two facts that are indeed confusing:
- `recommended-with-formatting` is used for projects _without_ Prettier formatting (but _with_ ESLint formatting checks)
- `eslint-config-prettier` disables existing rules rather than adding any new ones

Reply:
Thanks, and sorry